rfd/0208-proxy-desktop-session.md

```diff
@@ -0,0 +1,112 @@
+---
+authors: Grzegorz Zdunek (grzegorz.zdunek@goteleport.com)
+state: draft
+---
+
+# RFD 208 - Add ProxyDesktopSession API to Teleport Proxy Service
+
+## Required Approvals
+
+- Engineering: @zmb3 && (@rosstimothy || @espadolini)
+
+## What
+
+Introduce a new API in the Teleport Proxy Service to support streaming desktop sessions to Teleport Connect.
+
+## Why
+
+Remote desktop sessions in Teleport rely on the Teleport Desktop Protocol (TDP) for communication.
+To support interactive sessions, a bidirectional stream must be established between the client and the Windows Desktop Service to send and receive TDP messages in real time.
+
+## Details
+
+In the Web UI, the TDP stream is transmitted over a cookie-authenticated WebSocket. 
+
+However, Teleport Connect (which uses tsh under the hood) does not rely on cookies for authentication—instead, it uses client certificates.
+To support this scenario, a new gRPC-based API called `ProxyDesktopSession` will be added to the `TransportService` in the Teleport Proxy. 
+
+### Architecture
+
+```
+            gRPC bidi stream   
++-----+   (ProxyDesktopSession)    +----------------+      TCP connection     +-------------------------+
+| tsh | <=======================>  |  Proxy Service | <=====================> | Windows Desktop Service |
++-----+                            +----------------+                         +-------------------------+
+        <===================================================================>
+                              TLS connection (end-to-end)
+```
+
+`ProxyDesktopSession` won't transmit TDP messages directly.
+Instead, it will tunnel a TLS connection over gRPC, which will then carry the TDP messages.
+
+This design simplifies per-session MFA by allowing the client to present an MFA-verified certificate over a single TLS connection that terminates at the Windows Desktop Service.
```

Comment:
How does the client know that it needs to obtain a MFA-verified cert?

Reply:
I wanted to generate the cert with `TeleportClient.IssueUserCertsWithMFA`, which calls `IsMFARequired` under the hood.

```go
cert, err := clusterClient.IssueUserCertsWithMFA(ctx, client.ReissueParams{
	RouteToCluster: clusterClient.SiteName,
	TTL:            clusterClient.KeyTTL,
	RouteToDesktop: proto.RouteToWindowsDesktop{
		WindowsDesktop: desktopName,
		Login:          login,
	},
})
```

Reply:
Oh, we need a dedicated per-desktop cert anyway, I see.

Reply:
Yes, I didn't mention it in the RFD, but in tsh we will need to add support for windows desktops certs.